### PR TITLE
Sf/oc 2298

### PR DESCRIPTION
**Pull request by mmzyk**
@seth Pull request for user's chef_db work.

This is likely the most straight forward of the pull requests. Anyone else who should be included?

### CODE REVIEW

File: itest/itest.erl [hunk 1]
```diff
@@ -1094,6 +1133,42 @@ insert_node_data() ->
     ?assertEqual(Expected, Results).
 
 %%%======================================================================
+%%% USERS
+%%%======================================================================
+
```
Comment (seth) [inline itest/itest.erl:1146]: It would be good to add a test for the admin count functionality

File: priv/mysql_statements.config [hunk 2]
```diff
@@ -19,9 +19,39 @@
...
+      FROM osc_users WHERE username = ?">>}.
+
+{delete_user_by_username,
+  <<"DELETE FROM osc_users WHERE username = ?">>}.
+
+{list_users, <<"SELECT username FROM osc_users">>}.
+
+%% update query need to go here
 
 %% node queries
```
Comment (seth) [inline priv/mysql_statements.config:60]: I think the count admin query is missing for mysql?

File: src/chef_db.erl [hunk 3]
```diff
@@ -969,25 +999,25 @@ update_fun(#chef_role{}) ->
...
-    create_client;
-create_fun(#chef_node{}) ->
-    create_node;
-create_fun(#chef_role{}) ->
-    create_role;
-create_fun(#chef_sandbox{}) ->
-    create_sandbox;
-create_fun(#chef_cookbook_version{}) ->
-    create_cookbook_version.
+-spec create(chef_object() | #chef_user{} | #chef_sandbox{}, #context{}, object_id()) -> ok | {conflict, term()} | {error, term()}.
```
Comment (seth) [inline src/chef_db.erl:975]: This is a nice cleanup. :)

**Comment by mmzyk** [comment]
Late push addresses all the comments I saw here. cc/ @seth

File: src/chef_db.erl (hunk 3 above)
Comment (mmzyk) [inline src/chef_db.erl:975]: Thanks. I'd like to do more, but this is a start.

**Pull request merged**